### PR TITLE
@mczernek/expo sms/handle empty addresses array

### PR DESCRIPTION
**Pull request by mczernek**
# Why

THis fixes issue [#2673](https://github.com/expo/expo/issues/2673) on Android, and provides consistent behavior for sending SMSes with empty addresses list between Android and iOS.

# Test Plan

SMSScreen in ncl now supports this scenario.

### CODE REVIEW

File: packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java [hunk 1]
```diff
@@ -120,4 +110,16 @@ public void onHostPause() {
   public void onHostDestroy() {
     // do nothing
   }
+
+  private String constructSmsTo(List<String> addresses) {
```
Comment (sjchmiela) [inline packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java]:
```suggestion
  private String constructRecipients(List<String> addresses) {
```
?

**Review by sjchmiela — COMMENTED** [review]
Thanks for taking care of this, congratulations on your first pull request! 🎉

File: packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java [hunk 2]
```diff
@@ -120,4 +110,16 @@ public void onHostPause() {
...
+    if(addresses.size() > 0) {
```
Comment (tsapeta) [inline packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java]:
```suggestion
    if (addresses.size() > 0) {
```

File: packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java [hunk 3]
```diff
@@ -120,4 +110,16 @@ public void onHostPause() {
...
+      final StringBuilder addressesBuilder = new StringBuilder(addresses.get(0));
+      for (int idx = 1; idx < addresses.size(); idx++) {
```
Comment (tsapeta) [inline packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java]: `for (String address : addresses) {` might be better, we use this format in most places

**Review by tsapeta — CHANGES_REQUESTED** [review]
Nice! Congrats on the first PR 😉

File: packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java [hunk 4]
```diff
@@ -51,24 +52,13 @@ public void setModuleRegistry(ModuleRegistry moduleRegistry) {
...
   @ExpoMethod
   public void sendSMSAsync(final ArrayList<String> addresses, final String message, final Promise promise) {
-    if (!getContext().getPackageManager().hasSystemFeature(PackageManager.FEATURE_TELEPHONY) &&
-        !getContext().getPackageManager().hasSystemFeature(PackageManager.FEATURE_TELEPHONY_CDMA)) {
-      promise.reject(ERROR_TAG + "_UNAVAILABLE", "SMS service not available");
-      return;
-    }
-
-    if (mPromise != null) {
+    if (pendingPromise != null) {
```
[the remark this answers is not in the document]
Comment (mczernek) [reply packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java]: IMHO, it is redundant. 
It should not matter whether TELEPHONY is available, as long as there is any app willing to send SMSes. 
The worst can happen now is to open SMS app which in turn will not be able to sens SMS. I believe that it is more consistent and what users should really expect, especially on emulators, which might not have TELEPHONY, but will usually have SMS app.

On the other hand, it can cause some inconsistencies comparing to current behavior, so it might not be a good idea to change it now. It is up to your advice.

File: packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java [hunk 5]
```diff
@@ -120,4 +110,16 @@ public void onHostPause() {
   public void onHostDestroy() {
     // do nothing
   }
+
+  private String constructRecipients(List<String> addresses) {
+    if (addresses.size() > 0) {
+      final StringBuilder addressesBuilder = new StringBuilder(addresses.get(0));
+      for (String address: addresses) {
```
Comment (sjchmiela) [inline packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java]:
```suggestion
      for (String address : addresses) {
```

I would suggest configuring Android Studio to format files on save. It has some downsides, like there are files in the project that aren't formatted and editing them will lead to unnecessary changes, but apart from that it's the easiest way to make sure files are formatted properly.

File: packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java [hunk 6]
```diff
@@ -51,24 +52,13 @@ public void setModuleRegistry(ModuleRegistry moduleRegistry) {
...
   @ExpoMethod
   public void sendSMSAsync(final ArrayList<String> addresses, final String message, final Promise promise) {
-    if (!getContext().getPackageManager().hasSystemFeature(PackageManager.FEATURE_TELEPHONY) &&
-        !getContext().getPackageManager().hasSystemFeature(PackageManager.FEATURE_TELEPHONY_CDMA)) {
-      promise.reject(ERROR_TAG + "_UNAVAILABLE", "SMS service not available");
-      return;
-    }
-
-    if (mPromise != null) {
+    if (mPendingPromise != null) {
```
Comment (sjchmiela) [inline packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java]: Could you please also add CHANGELOG entry for the bug you fixed and that promise won't be rejected if "there's no telephony feature" on the device?

File: CHANGELOG.md [hunk 7]
```diff
@@ -30,12 +30,13 @@ This is the log of notable changes to the Expo client that are developer-facing.
...
+- fixed several issues related to `expo-av` by [@Szymon20000](https://github.com/Szymon20000) ([#3539](https://github.com/expo/expo/pull/3539))
 - `Location.getCurrentPositionAsync` and `Location.watchPositionAsync` are now automatically asking for high accuracy location services by [@tsapeta](https://github.com/tsapeta) ([#3273](https://github.com/expo/expo/pull/3273))
 - fix `Location.getCurrentPositionAsync` hanging on simultaneous calls by [@tsapeta](https://github.com/tsapeta) ([#3273](https://github.com/expo/expo/pull/3273))
 - fix `ImagePicker.launchImageLibraryAsync` and `ImageManipulator.manipulateAsync` in SDKs lower than 32 [@bbarthec](https://github.com/bbarthec) ([#3159](https://github.com/expo/expo/pull/3159))
 - fix app crash when attempting to `console.log(Object.create(null))` by [@juangl](https://github.com/juangl) ([#3143](https://github.com/expo/expo/pull/3143))
 - `GoogleSignInOptions.offlineAccess` is now corrected `GoogleSignInOptions.isOfflineEnabled`
+- fix `SMS.sendSMSAsync` crash on android for empty addresses list. Promise won't be rejected when there is no TELEPHONY feature on Android device, only when there is no SMS app.
```
Comment (sjchmiela) [inline CHANGELOG.md]:
```suggestion
- fix `SMS.sendSMSAsync` crash on android for empty addresses list. Promise won't be rejected when there is no TELEPHONY feature on Android device, only when there is no SMS app by [@mczernek](https://github.com/mczernek) ([#3656](https://github.com/expo/expo/pull/3656))
```

**Review by sjchmiela — APPROVED** [review]
